src/providers/go-to-definition.ts

```diff
@@ -0,0 +1,23 @@
+import * as vscode from "vscode";
+
+export class MF6DefinitionProvider implements vscode.DefinitionProvider {
+  public async provideDefinition(
+    document: vscode.TextDocument,
+    position: vscode.Position,
+  ): Promise<vscode.Location | vscode.Location[]> {
+    const wordRange = document.getWordRangeAtPosition(position);
+    if (!wordRange) {
+      return [];
+    }
+    const word = document.getText(wordRange);
+    const fileUri = vscode.Uri.joinPath(document.uri, "..", word);
+
+    try {
+      await vscode.workspace.fs.stat(fileUri);
+      return new vscode.Location(fileUri, new vscode.Position(0, 0));
+    } catch (error) {
+      vscode.window.showWarningMessage(`File ${word} not found`);
+      return [];
+    }
```

Comment:
I would extract a block of code to check if file exists for better readability of the code. You may put it in a utils module or just outside of `MF6DefinitionProvider` definition.

```suggestion
    if (!checkFileExists(fileUri)) {
        vscode.window.showWarningMessage(`File ${word} not found`);
        return null;
    }
    return new vscode.Location(fileUri, new vscode.Position(0, 0));
```

The following can be the definition of `checkFileExists`.

```js
async function checkFileExists(uri: vscode.Uri) {
  try {
    await vscode.workspace.fs.stat(uri);
    return true;
  } catch {
    return false;
  }
}
```

Reply:
Tried adapting these changes here: https://github.com/martclanor/vscode-mf6-syntax/pull/40/commits/b4d35dc7b38131145533ceb9bb0e472a20869bef.

However, I notice a difference in behavior if the file does not exist. Before, VS Code would show smth like a tooltip saying that the file does not exist, 
<img width="221" alt="image" src="https://github.com/user-attachments/assets/fa1cd323-511c-4d7d-b778-296509a77c29" />

but after the changes, VS Code opens up a new tab, with an error saying that file does not exist which I find a bit more intrusive. 
![image](https://github.com/user-attachments/assets/eb7bcb3e-a8ad-4548-bce6-9a92ca1fe937)


Not sure yet what made this difference. Thoughts on this?

Reply:
Sorry, you should await the check.

it should be `if (!(await checkFileExists(fileUri))`.